platform/core.windows/src/org/netbeans/core/windows/options/LafPanel.java

```diff
@@ -202,8 +203,19 @@ boolean valid() {
 
     private void initLookAndFeel() {
         lafs.clear();
-        for( LookAndFeelInfo i : UIManager.getInstalledLookAndFeels() ) {
-            lafs.add( i );
+        String lafFilter = NbBundle.getMessage(LafPanel.class, "Laf_Filter").trim();
+        if (lafFilter.isEmpty()) {
+            lafs.addAll(Arrays.asList(UIManager.getInstalledLookAndFeels()));
+        } else {
+            LookAndFeelInfo cur = getCurrentLaF();
+            LookAndFeelInfo pref = getPreferredLaF();
+            for (LookAndFeelInfo li : UIManager.getInstalledLookAndFeels()) {
+                String name = li.getClassName();
+                name = name.substring(name.lastIndexOf('.') + 1, name.length());
+                if (li == cur || li == pref || !lafFilter.contains(name)) {
```

Comment:
Not sure if LookAndFeelInfo can be compared by object identity; probably better to use LookAndFeelInfo.getClassName().equals(...) (but be careful with null values).

Reply:
Using String.contains could lead to surprising effects here. Maybe better to use Set<String>.contains

Reply:
In my first iteration they were guaranteed to be the same instances because they were cached in this class.  I guess safer to do the class name equality now, although they are the same instances returned.

String contains should be fine here IMO, if a little lenient, and removes the need to parse the input into a Set.